jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.cpp

```diff
@@ -42,6 +43,53 @@
  */
 
 /**
+ * Insert leading zeros to a string of a number if needed.
+ */
+static void
+ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< input/output string */
+                                ecma_number_t num, /**< input number */
+                                uint32_t length) /**< length of string of number */
+{
+  JERRY_ASSERT (length >= 1);
+
+  /* If the length is bigger than the number of digits in num, then insert leding zeros. */
+  for (uint32_t i = length - 1; i > 0 && num < pow (10,i); i--)
+  {
+    ecma_string_t *zero_str_p = ecma_new_ecma_string_from_uint32 (0);
+    ecma_string_t *concat_p = ecma_concat_ecma_strings (zero_str_p, *str_p);
+    ecma_deref_ecma_string (zero_str_p);
+    ecma_deref_ecma_string (*str_p);
+    *str_p = concat_p;
+  }
+} /* ecma_date_insert_leading_zeros */
+
+/**
+ * Insert a number to the start of a string with a specific separator character and
+ * fix length. If the length is bigger than the number of digits in num, then insert leding zeros.
+ */
+static void
+ecma_date_insert_num_with_sep (ecma_string_t **str_p, /**< input/output string */
+                               ecma_number_t num, /**< input number */
+                               lit_magic_string_id_t magic_str_id, /**< separator character id */
+                               uint32_t length) /**< length of string of number */
+{
+  ecma_string_t *magic_string_p = ecma_get_magic_string (magic_str_id);
+
+  ecma_string_t *concat_p = ecma_concat_ecma_strings (magic_string_p, *str_p);
+  ecma_deref_ecma_string (magic_string_p);
+  ecma_deref_ecma_string (*str_p);
+  *str_p = concat_p;
+
+  ecma_string_t *num_str_p = ecma_new_ecma_string_from_number (num);
+  concat_p = ecma_concat_ecma_strings (num_str_p, *str_p);
+  ecma_deref_ecma_string (num_str_p);
+  ecma_deref_ecma_string (*str_p);
+  *str_p = concat_p;
```

Comment:
@LaszloLango, I see. I think that the approach that you've described is good for now, until the hidden dereference issue wouldn't be solved somehow (maybe, after switching to C++ in the whole engine, it would be solved in a simple way).
The code is ok for me.